### ISSUE TITLE
`includeOriginalRequest=false` does not work for the error responses.

### ISSUE THREAD

**Issue by ahmetoz** [commercetools/commercetools-sdk-typescript#240]
There is a param `includeOriginalRequest` on the `HttpMiddlewareOptions`, but this is working only for the successful response, I wonder why this is not considered as a check on the error responses, what is the specific reason for it ?

Use case: Filter the request details from the logs as it could contain some sensitive information such as email, address of the customer.

From the docs: https://commercetools.github.io/nodejs/sdk/api/sdkMiddlewareHttp.html
````
includeOriginalRequest (Boolean): flag whether to include the original request sent in the response. Can be useful if you want to see the final request being sent. 
````
On source code:
https://github.com/commercetools/commercetools-sdk-typescript/blob/650e5766262ffb1bf3e1f123239c2be4fa254ed9/packages/sdk-client/src/sdk-middleware-http/http.ts#L205-L214

**Comment by ajimae** [comment]
Hi ahmetoz,

What do you want as the expected behaviour hear?

On error, the originalRequest shouldn't be included or on error it should be included with the sensitive details filtered out?

**Comment by ahmetoz** [comment]
> What do you want as the expected behaviour here?
Currently this flag is working only for the success/ok responses, so I expect it would work for the error too, I didn't check the other flags but might be the same case if they are only relying on the success response.

**Comment by ajimae** [comment]
Yes, I will implement it in the coming week.

**Comment by ajimae** [comment]
> Hi @ajimae any decision for this task ? Will it be implemented ?

Hi @ahmetoz 

I was looking to implement this feature but I found out it's the default behavior for error requests.

Look at this from the http source code.

https://github.com/commercetools/commercetools-sdk-typescript/blob/c01e0e4f67a3e40601a7968a731737fc78cfa02f/packages/sdk-client/src/sdk-middleware-http/http.ts#L247

https://github.com/commercetools/commercetools-sdk-typescript/blob/c01e0e4f67a3e40601a7968a731737fc78cfa02f/packages/sdk-client/src/sdk-middleware-http/http.ts#L305

The only difference here is that we pass in a boolean to mask out (redact) sensitive request data.

So, I don't think we still need to check for `includeOriginalRequest` on error.

**Comment by ahmetoz** [comment]
Assume the request is a create payment and the error happened the request will be logged, which will contain all payload, not sure what you meant by it's already implemented. Can you give an example ?

**Comment by ajimae** [comment]
Hi @ahmetoz 

An example will be to trigger an error while a request is still being processed. You can throw an error just as a request is being processed or you ca use this configuration I crafted just for this purpose, essentially, I aborted the request midway into the request process to get project details.

```js
import AbortController from 'abort-controller'
.
.
.
const httpMiddlewareOptions = {
  host: 'https://api.europe-west1.gcp.commercetools.com',
  timeout: 10,
  getAbortController: () => new AbortController(),
  fetch,
}
```

This is the log I got after firing up a request using this `httpMiddlewareOptions` configuration.

```bash
NetworkError: The user aborted a request.
    at /Users/xxxxxx/workspace/commercetools-sdk-typescript/node_modules/@commercetools/sdk-middleware-http/dist/sdk-middleware-http.cjs.js:452:23
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  code: 0,
  statusCode: 0,
  status: 0,
  originalRequest: {
    baseUri: 'https://api.europe-west1.gcp.commercetools.com',
    method: 'GET',
    uriTemplate: '/{projectKey}',
    pathVariables: { projectKey: 'xxxxxx' },
    headers: { Authorization: 'Bearer ********' },
    uri: '/xxxxxx'
  },
  retryCount: 0
}
``` 

As you can see, the `originalRequest` is part of the error response as discussed earlier.

**Comment by ahmetoz** [comment]
> As you can see, the originalRequest is part of the error response as discussed earlier.

Maybe it should not, when you set `includeOriginalRequest=false` ? Sorry did I explain it different in the beginning, of course the response will contain the original request.

**Comment by ajimae** [comment]
> Maybe it should not, when you set `includeOriginalRequest=false` ? Sorry did I explain it different in the beginning, of course the response will contain the original request when the flag is true (default is true) and when the flag is false it should not, as explained before this flag behaves only for successful requests.

So the SDK works just as you have explained by default however, for error responses the `originalRequest` is always included and sensitive data are also redacted by default also. If you don't want the sensitive to be redacted then you can pass in an option to the `httpMiddlewareOption` called `maskSensitiveHeaderData` and set its value to false.

See [this](https://github.com/commercetools/commercetools-sdk-typescript/blob/000f3f728500f36a71ccc24e91ee7160aad8bfd1/packages/sdk-client/src/sdk-middleware-http/http.ts#L51-L61) and [this](https://github.com/commercetools/commercetools-sdk-typescript/blob/c01e0e4f67a3e40601a7968a731737fc78cfa02f/packages/sdk-client/src/sdk-middleware-http/http.ts#L308)

Also, if you have a sensitive data that is included in the header that is not redacted then you can let me know so I can extend this functionality to include custom header keys to be redacted.

**Comment by butenkor** [comment]
Hi @ajimae , it is not about header but the originalRequest as it might contain sensitive information like customer data which runs under GDPR law. Such data should not be logged on error and thus when the flag includeOriginalRequest=false is set then it should not return it to the SDK users in case of the error. Besides that it is also not consistent in comparison how the flag behaves in success cases.

**Comment by ajimae** [comment]
Hi @butenkor 

Thanks for your input, I will raise a PR to effect this change as soon as I can.
